### PR TITLE
operator-sdk/internal/generate: fix CSV path segments for multi-field structs

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**

Fixes a bug that causes path segments to inadvertently re-use memory from other path segments, causing `specDescriptors` and `statusDescriptors` fields to end up with incorrect paths.

**Motivation for the change:**

Fix bug = good :)

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: internal/generate/clusterserviceversion/bases/definitions/ast.go [hunk 1]
```diff
@@ -66,9 +66,11 @@ func (g generator) getMarkedChildrenOfField(root markers.FieldInfo) (map[string]
 						}
 						// Create a new set of path segments using the parent's segments
 						// and add the field to the next fields to search.
+						parentSegments := make([]string, len(field.pathSegments))
```
Comment (estroz) [inline internal/generate/clusterserviceversion/bases/definitions/ast.go:69]: `append` won't create a new underlying array if its cap is sufficient
```suggestion
						parentSegments := make([]string, len(field.pathSegments), len(field.pathSegments)+1)
```

**Review by estroz — APPROVED** [review]
LGTM after addressing nit

**Review by camilamacedo86 — APPROVED** [review]
/approve

Could we not add these markers in the samples?

**Comment by joelanford** [comment]
@camilamacedo86 :+1: to adding some spec and/or status descriptor markers to our samples, especially if we don't have to make any other major changes in the samples to support adding them. Can you make another issue to track this?

**Pull request merged**